### PR TITLE
Add ability to set pod annotations

### LINKED ISSUE #177: Add support for passing Deployment's pod spec annotations
Currently, there is an option to pass annotations to service but not to pods. 
Some system services like kube2ami etc are using pod's spec annotations.

apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: nginx-deployment
spec:
  replicas: 3
  template:
    metadata:
      annotations:
        iam.amazonaws.com/role: role-arn

### PR DESCRIPTION
**Pull request by chrisjs**
fixes #177

### CODE REVIEW

**Comment by chrisjs** [comment]
tested via:

stream create --name ticktock --definition "time | log"
stream deploy ticktock --properties "deployer.log.kubernetes.podAnnotations=iam.amazonaws.com/role: myarn"

kubectl get --raw /apis/apps/v1/namespaces/default/deployments/ticktock-log | python -m json.tool

and verified:

```
{
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": {
      ....
    },
    "spec": {
        "replicas": 1,
        ....
        "template": {
            "metadata": {
                "annotations": {
                    "iam.amazonaws.com/role": "myarn"
                },
                ....
            },
```

**Comment by shahbour** [comment]
I suggest adding the same to [KubernetesTaskLauncher](https://github.com/shahbour/spring-cloud-deployer-kubernetes/commit/280ddabc0ea85fcb0ac79a61e53cd264155fb02f) .  We did in our environment to correctly parse the logs using fluent-bit

**Comment by chrisjs** [comment]
@shahbour Thanks - I've created https://github.com/spring-cloud/spring-cloud-deployer-kubernetes/issues/204 to track that specific enhancement

File: src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesAppDeployer.java [hunk 1]
```diff
@@ -424,34 +424,50 @@ protected void createService(String appId, AppDeploymentRequest request, Map<Str
...
+	 * Extracts annotations from the provided value
+	 *
+	 * @param annotation The deployment request annotations
+	 * @return {@link Map} of annotations
+	 */
+	private Map<String, String> getAnnotations(String annotation) {
+		Map<String, String> annotations = new HashMap<>();
+
+		if (StringUtils.hasText(annotation)) {
+			String[] annotationPairs = annotation.split(",");
```
Comment (ilayaperumalg) [inline src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesAppDeployer.java]: I guess we could add some unit tests on how this comma separated annotations are retrieved as Map.

Comment (ilayaperumalg) [reply src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesAppDeployer.java]: Perhaps we could conveniently add that as part of the existing integration test. I can add it when merging.

File: src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesDeployerProperties.java [hunk 2]
```diff
@@ -210,6 +210,11 @@ public void setStorageClassName(String storageClassName) {
 	 */
 	private String serviceAnnotations = null;
 
+	/**
+	 * Pod annotations to set for the pod created for each deployment.
+	 */
+	private String podAnnotations;
```
Comment (ilayaperumalg) [inline src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesDeployerProperties.java]: I think we need a doc update for this (and for service annotations as well). Created https://github.com/spring-cloud/spring-cloud-deployer-kubernetes/issues/205

**Comment by ilayaperumalg** [comment]
LGTM, verified the changes on minikube and the annotations on the pod spec looks good as well. 

Merging with the test update mentioned in the comments. Created a separate issue to track doc update.

**Comment by ilayaperumalg** [comment]
Merged as 048c539f69c3fdad95bc5a2d642591144d1c765a along with the test updates 09838cbdacbce2f7aeb127364fd3444f110cbaaf

**Pull request closed without merging**